recipes/vtk/all/conanfile.py

```diff
@@ -0,0 +1,1418 @@
+        # TODO try VTK_USE_VIDEO_FOR_WINDOWS   for video capture
+        # TODO try VTK_USE_VIDEO_FOR_WINDOWS_CAPTURE   for video capture
+        # TODO try VTK_USE_MICROSOFT_MEDIA_FOUNDATION   for video capture (MP4)
+
+
+        # this little function only to help mark out the special
+        # default/yes/no/want/dont_want options
+        def _yesno(flag):
+            return flag
+
+
+        # groups can be:  DEFAULT   DONT_WANT   WANT   YES   NO
+        # Note that YES is like WANT, but will show errors if can't make everything
+        # NO is also more forceful than DONT_WANT
+        # Default is DONT_WANT, let it auto-enable when required
+        tc.variables["VTK_GROUP_ENABLE_Imaging"]    = _yesno(self.options.group_enable_Imaging)    # TODO test
+        tc.variables["VTK_GROUP_ENABLE_MPI"]        = _yesno(self.options.group_enable_MPI)        # TODO test
+        tc.variables["VTK_GROUP_ENABLE_Rendering"]  = _yesno(self.options.group_enable_Rendering)
+        tc.variables["VTK_GROUP_ENABLE_StandAlone"] = _yesno(self.options.group_enable_StandAlone)  # TODO test
+        tc.variables["VTK_GROUP_ENABLE_Views"]      = _yesno(self.options.group_enable_Views)       # TODO test
+        tc.variables["VTK_GROUP_ENABLE_Web"]        = _yesno(self.options.group_enable_Web)         # TODO test
+
+        # for Qt, can also use the more specific MODULE options below
+        tc.variables["VTK_GROUP_ENABLE_Qt"] = _yesno(self.options.group_enable_Qt)
+
+        ##### QT ######
+        # QT has a few modules, we'll be specific
+        tc.variables["VTK_QT_VERSION"]                          = self.options.qt_version
+
+        # Setup ALL our discovered modules
+        # this will generate lots of variables, such as:
+        # tc.variables["VTK_MODULE_ENABLE_VTK_RenderingCore"]     = _yesno(self.options.module_enable_RenderingCore)
+        for mod in vtk_all_modules:
+            option_name   = f"module_enable_{mod}"
+            variable_name = f"VTK_MODULE_ENABLE_VTK_{mod}"
+            tc.variables[variable_name] = _yesno(self.options.get_safe(option_name))
+            # print(f"Added variable {variable_name} = {_yesno(self.options.get_safe(option_name))}")
+
+        # TODO if true (or all) then system has to install postgres dev package
+
+        ##### SMP parallelism ####  Sequential  STDThread  OpenMP  TBB
+        # Note that STDThread seems to be available by default
+        tc.variables["VTK_SMP_IMPLEMENTATION_TYPE"] = self.options.smp_implementation_type
+        # Change change the mode during runtime, if you enable the backends like so:
+        tc.variables["VTK_SMP_ENABLE_Sequential"]   = self.options.smp_enable_Sequential
+        tc.variables["VTK_SMP_ENABLE_STDThread"]    = self.options.smp_enable_STDThread
+        tc.variables["VTK_SMP_ENABLE_OpenMP"]       = self.options.smp_enable_OpenMP
+        tc.variables["VTK_SMP_ENABLE_TBB"]          = self.options.smp_enable_TBB
```

Comment:
TBB option should also lead to adding onetbb to "requires" and to dependencies of CommonCore

Reply:
Hmm probably, I'm not enabling SMP at the moment so it would be good if you could test that in future.